src/o365/spo/commands/listitem/listitem-record-declare.ts

```diff
@@ -0,0 +1,309 @@
+
+        listId = res.Id;
+        const requestBody = this.generateDeclareRecordRequestBody(webIdentity, listId, id, dateArgument);
+
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_vti_bin/client.svc/ProcessQuery`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'Content-Type': 'text/xml',
+            'X-RequestDigest': formDigestValue
+          }),
+          body: requestBody
+        };
+
+        if (this.debug) {
+          cmd.log("Executing declare item as record web request...");
+          cmd.log(requestOptions);
+          cmd.log("");
+          cmd.log("Body:");
+          cmd.log(requestBody);
+          cmd.log("");
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: string): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        const json: ClientSvcResponse = JSON.parse(res);
+        const response: ClientSvcResponseContents = json[0];
+
+        if (response.ErrorInfo) {
+          cmd.log(new CommandError(response.ErrorInfo.ErrorMessage));
+        }
+        else {
+          const result: boolean = json[json.length - 1];
+          cmd.log(result);
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedPromise(err, cmd, cb));
+  }
+
+  protected generateDeclareRecordRequestBody(webIdentity: string, listId: string, id: string, date: string): string {
+
+    let requestBody = "";
+    if (date.length == 10) {
+      requestBody = `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><StaticMethod TypeId="{ea8e1356-5910-4e69-bc05-d0c30ed657fc}" Name="DeclareItemAsRecordWithDeclarationDate" Id="48"><Parameters><Parameter ObjectPathId="21" /><Parameter Type="DateTime">${date}</Parameter></Parameters></StaticMethod></Actions><ObjectPaths><Identity Id="21" Name="${webIdentity}:list:${listId}:item:${id},1" /></ObjectPaths></Request>`;
+    }
+    else {
+      requestBody = `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><StaticMethod TypeId="{ea8e1356-5910-4e69-bc05-d0c30ed657fc}" Name="DeclareItemAsRecord" Id="37"><Parameters><Parameter ObjectPathId="12" /></Parameters></StaticMethod></Actions><ObjectPaths><Identity Id="12" Name="${webIdentity}:list:${listId}:item:${id},1" /></ObjectPaths></Request>`;
+    }
+
+    return requestBody;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: "-u, --webUrl <webUrl>",
+        description: "URL of the site where the item should be added"
```

Comment:
Seems like the descriptions of the options still refer to the command from where you copied them. We should update them following the spec: https://github.com/pnp/office365-cli/issues/769

Reply:
Updated!